sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py

```diff
@@ -171,11 +193,20 @@ def get_schema_id(self, schema_group, schema_name, serialization_type, schema_co
         except AttributeError:
             pass
 
-        return self._generated_client.schema.query_id_by_content(
-            group_name=schema_group,
-            schema_name=schema_name,
-            schema_content=schema_content,
-            x_schema_type=serialization_type,
-            cls=_parse_response_schema_id,
-            **kwargs
-        )
+        try:
+            return self._description_to_properties[
+                (schema_group, schema_name, serialization_type, schema_content)
+            ]
+        except KeyError:
+            schema_properties = self._generated_client.schema.query_id_by_content(
+                group_name=schema_group,
+                schema_name=schema_name,
+                schema_content=schema_content,
+                x_schema_type=serialization_type,
+                cls=_parse_response_schema_id,
+                **kwargs
+            )
+            self._description_to_properties[
+                (schema_group, schema_name, serialization_type, schema_content)
+            ] = schema_properties
```

Comment:
same comment as the `get_schema` method, we could also update `self._id_to_schema` here?

Reply:
in this case, I think it makes sense to also update the other dict!